lexer.html

```diff
@@ -0,0 +1,140 @@
+<link rel="import" href="common.html">
+<script>
+function ShadeyCssLexer() {
+  'use strict';
+  // NOTE(cdata): Generally, tokens are structured as three-valued tuple,
```

Comment:
I will change to using an object in favor of a tuple, but I'm going to keep the factory pattern in favor of using a constructor via new. The factory pattern appears to be a strong performance choice across the board for different browsers: http://jsperf.com/create-objects-in-different-ways/3